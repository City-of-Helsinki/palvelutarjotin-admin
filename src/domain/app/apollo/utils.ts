import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
