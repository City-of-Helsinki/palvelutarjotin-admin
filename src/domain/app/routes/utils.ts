import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
