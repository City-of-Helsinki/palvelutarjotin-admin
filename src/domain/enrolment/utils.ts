import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
