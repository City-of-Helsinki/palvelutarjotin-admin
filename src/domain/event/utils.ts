import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
