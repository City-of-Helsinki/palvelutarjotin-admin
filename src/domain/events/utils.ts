import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
