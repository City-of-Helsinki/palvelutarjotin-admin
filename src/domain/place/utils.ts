/**
 * Generate link to get Hsl directions
 * @param {string} streetAddress
 * @param {string} addressLocality
 * @return {string}
 */
export const generateHslLink = (
  streetAddress: string,
  addressLocality: string
) => {
  return `http://www.reittiopas.fi/fi/?to=${streetAddress.replace(
    /\s+/g,
    '+'
  )},${addressLocality}`;
};

/**
 * Generate link to service map
 * @param {string} id
 * @return {string}
 */
export const generateServiceMapLink = (id: string) => {
  const unitId = id.split(':')[1];
  return `https://palvelukartta.hel.fi/fi/unit/${unitId}`;
};
