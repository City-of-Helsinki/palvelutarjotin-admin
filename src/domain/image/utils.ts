import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
