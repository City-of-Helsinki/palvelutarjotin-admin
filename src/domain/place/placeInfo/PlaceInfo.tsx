import * as React from 'react';
import { useTranslation } from 'react-i18next';

import { usePlaceQuery } from '../../../generated/graphql';
import { Language } from '../../../types';
import getLocalizedString from '../../../utils/getLocalizedString';
import VenueInfo from '../../venue/venueInfo/VenueInfo';
import { generateHslLink, generateServiceMapLink } from '../utils';
import styles from './placeInfo.module.scss';

interface Props {
  id: string;
  language: Language;
  onEditButtonClick?: (show: boolean) => void;
  showEditButton?: boolean;
  showVenueInfo?: boolean;
}

const PlaceInfo: React.FC<Props> = ({
  id,
  language,
  onEditButtonClick,
  showEditButton,
  showVenueInfo,
}) => {
  const { t } = useTranslation();
  const { data } = usePlaceQuery({ variables: { id }, skip: !id });

  if (!data) return null;

  const name = getLocalizedString(data.place?.name || {}, language);
  const streetAddress = getLocalizedString(
    data.place?.streetAddress || {},
    language
  );
  const telephone = getLocalizedString(data.place?.telephone || {}, language);
  const addressLocality = getLocalizedString(
    data.place?.addressLocality || {},
    language
  );
  const serviceMapLink = generateServiceMapLink(id, language);
  const hslLink = generateHslLink(streetAddress, addressLocality, language);

  const handleEditButtonClick = () => {
    if (onEditButtonClick) {
      onEditButtonClick(true);
    }
  };

  return (
    <div className={styles.placeInfo}>
      {name && (
        <div>
          {name}{' '}
          {showEditButton && (
            <button
              className={styles.buttonChange}
              onClick={handleEditButtonClick}
              type="button"
            >
              {t('form.placeInfo.buttonEdit')}
            </button>
          )}
        </div>
      )}
      {streetAddress && <p>{streetAddress}</p>}
      {telephone && <p>{telephone}</p>}
      <div className={styles.linkRow}>
        <div className={styles.linkTitle}>
          {t('form.placeInfo.labelServicemap')}
        </div>
        <a href={serviceMapLink} rel="noopener noreferrer" target="_blank">
          {serviceMapLink}
        </a>
      </div>
      <div className={styles.linkRow}>
        <div className={styles.linkTitle}>
          {t('form.placeInfo.labelReittiopas')}
        </div>
        <a href={hslLink} rel="noopener noreferrer" target="_blank">
          {hslLink}
        </a>
      </div>

      {showVenueInfo && <VenueInfo language={language} placeId={id} />}
    </div>
  );
};

export default PlaceInfo;
