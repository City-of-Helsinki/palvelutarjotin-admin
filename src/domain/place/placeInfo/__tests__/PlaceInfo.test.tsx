import * as React from 'react';

import { PlaceDocument } from '../../../../generated/graphql';
import {
  fakeLocalizedObject,
  fakePlace,
} from '../../../../utils/mockDataUtils';
import { render, screen, waitFor } from '../../../../utils/testUtils';
import PlaceInfo from '../PlaceInfo';

const placeResult = {
  data: {
    place: fakePlace({
      name: fakeLocalizedObject('Sellon kirjasto'),
      streetAddress: fakeLocalizedObject('LeppÃ¤vaarankatu 9'),
      telephone: fakeLocalizedObject('123321123'),
      addressLocality: fakeLocalizedObject('testi'),
    }),
  },
};

const placeId = placeResult.data.place.id;

const mocks = [
  {
    request: {
      query: PlaceDocument,
      variables: { id: placeId },
    },
    result: placeResult,
  },
];

test('PlaceInfo should match snapshot', async () => {
  const { container } = render(<PlaceInfo language="fi" id={placeId} />, {
    mocks,
  });

  await waitFor(() => {
    expect(screen.queryByText('Sellon kirjasto')).toBeInTheDocument();
  });

  expect(container.firstChild).toMatchSnapshot();
});
