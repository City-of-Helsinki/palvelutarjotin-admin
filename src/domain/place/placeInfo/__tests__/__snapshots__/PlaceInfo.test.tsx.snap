// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceInfo should match snapshot 1`] = `
<div
  class="placeInfo"
>
  <div>
    Sellon kirjasto
     
  </div>
  <p>
    Leppävaarankatu 9
  </p>
  <p>
    123321123
  </p>
  <div
    class="linkRow"
  >
    <div
      class="linkTitle"
    >
      Palvelukartta:
    </div>
    <a
      href="https://palvelukartta.hel.fi/fi/unit/undefined"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://palvelukartta.hel.fi/fi/unit/undefined
    </a>
  </div>
  <div
    class="linkRow"
  >
    <div
      class="linkTitle"
    >
      Reittiopas:
    </div>
    <a
      href="https://www.reittiopas.fi/fi/?to=Leppävaarankatu+9,testi"
      rel="noopener noreferrer"
      target="_blank"
    >
      https://www.reittiopas.fi/fi/?to=Leppävaarankatu+9,testi
    </a>
  </div>
</div>
`;
